src/core.jl

```diff
@@ -0,0 +1,170 @@
+"""
+Type representing any object drawable on image
+"""
+abstract Drawable
+
+"""
+    p = Point(x,y)
+    p = Point(c)
+
+A `Drawable` point on the image
+"""
+immutable Point <: Drawable
+    x::Int
+    y::Int
+end
+
+abstract Line <: Drawable
+abstract Circle <: Drawable
+
+
+"""
+    line = LineTwoPoints(p1, p2)
+
+A `Drawable` infinite length line passing through the two points
+`p1` and `p2`.
+"""
+immutable LineTwoPoints <: Line
+    p1::Point
+    p2::Point
+end
+
+"""
+    line = LineNormal(ρ, θ)
+
+A `Drawable` infinte length line having perpendicular length `ρ` from
+origin and angle `θ` between the perpendicular and x-axis
+"""
+immutable LineNormal <: Line
+    ρ::Real
+    θ::Real
```

Comment:
Yes, indeed it gave about 10 times performance boost (mean time). I will do the same for `Ellipse`, `CirclePointRadius` etc.
```julia
julia> immutable LineNormal
           ρ::Real
           θ::Real
       end

julia> immutable LineNormalParam{T<:Real, U<:Real}
           ρ::T
           θ::U
       end

julia> @benchmark LineNormal(1.0,1.0)
BenchmarkTools.Trial: 
  memory estimate:  32 bytes
  allocs estimate:  1
  --------------
  minimum time:     8.813 ns (0.00% GC)
  median time:      11.701 ns (0.00% GC)
  mean time:        13.548 ns (14.39% GC)
  maximum time:     1.697 μs (98.15% GC)
  --------------
  samples:          10000
  evals/sample:     999
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> @benchmark LineNormalParam(1.0,1.0)
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     1.888 ns (0.00% GC)
  median time:      1.892 ns (0.00% GC)
  mean time:        1.984 ns (0.00% GC)
  maximum time:     539.526 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000
  time tolerance:   5.00%
  memory tolerance: 1.00%
```
But why doesn't `@code_warntype` give any information about it?

```julia
julia> @code_warntype LineNormal(1.0,1.0)
Variables:
  #self#::Type{LineNormal}
  ρ::Float64
  θ::Float64

Body:
  begin 
      return $(Expr(:new, :(Main.LineNormal), :(ρ), :(θ)))
  end::LineNormal
```